source/slang/slang-ir-spirv-legalize.cpp

```diff
@@ -1612,6 +1612,41 @@ struct SPIRVLegalizationContext : public SourceEmitterBase
         }
     }
 
+    void legalizeSPIRVLinkageFunc(IRFunc* func)
+    {
+        // Check the `HLSLExport` decoration or `DownstreamExport` decoration. If any performs, we
+        // assume this function is exported, add SpirV export decoration and add linkage attribute
+        // later.
+        auto hlslExportDecor = func->findDecoration<IRHLSLExportDecoration>();
+        auto downstreamExportDecor = func->findDecoration<IRDownstreamModuleExportDecoration>();
+        if (hlslExportDecor || downstreamExportDecor)
+        {
+            IRBuilder builder(func);
+            builder.addDecorationIfNotExist(func, kIROp_SpirVExportDecoration);
+            return;
+        }
+        // Check the `UserExtern` decoration or `DownstreamImport` decoration. If any performs, we
+        // assume this function is imported, add SpirV extern decoration and add linkage attribute
+        // later.
```

Comment:
It seems like "If any performs" is a grammar error of some kind here. From context it seems like the intention is something like "If either is found."